engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -787,6 +787,24 @@ object Main {
     runMain(mainModule, Some(file), additionalArgs)
   }
 
+  /** If the stack trace ends with `Assertion_Error`, drops the internal assertion frames.
+    */
+  private def dropAssertionError(
+    stack: List[PolyglotException#StackFrame]
+  ): List[PolyglotException#StackFrame] = {
+    if (
+      stack.size > 1 && stack(0).getRootName.contains("Panic.throw") && stack(
+        1
+      ).getRootName.contains("Runtime.assert")
+    ) {
+      stack
+        .slice(1, stack.size)
+        .dropWhile(_.getRootName.contains("Runtime.assert"))
+    } else {
+      stack
+    }
+  }
```

Comment:
This only affects the 'uncaught' exception message printed in the runner.

It will not affect how the stack trace is displayed e.g. in the IDE or in the test runner, which fetch the stack trace in a different way.